facebook_connector.py

```diff
@@ -45,8 +53,23 @@ def resolve_user_id(user_id):
     Returns:
         The name of the user if it is able to find it, otherwise the input
     """
-    # TODO (neitsch): This method is not implemented yet.  Use the user_id
-    # (format: 12345678@facebook.com) to get the actual name of the user
+    fb_id_pattern_match = _FB_ID_PATTERN.match(user_id)
+    if fb_id_pattern_match:
+        fb_numeric_id = fb_id_pattern_match.group(1)
+        if fb_numeric_id not in _mapped_fb_ids:
+            username = fb_numeric_id
+            try:
+                fb_user_page = urlopen("https://www.facebook.com/{}".format(fb_numeric_id))
+                fb_page_title = BeautifulSoup(fb_user_page.read()).title.string
+                possible_username = fb_page_title.split("|")[0].strip()
+                if not (username.startswith('Security Check Required') or username.startswith('Page Not Found')):
```

Comment:
Flip this too so you're checking if it does start with that and break out early, it'll be clearer that way.